src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java

```diff
@@ -1,14 +1,53 @@
 package uk.nhs.digital.nhsconnect.nhais.service;
 
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Interchange;
+
+import org.hl7.fhir.r4.model.Organization;
 import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Practitioner;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Resource;
 import org.springframework.stereotype.Component;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.Interchange;
 
 @Component
 public class EdifactToFhirService {
 
-    Parameters convertToFhir(Interchange interchange) {
-        return new Parameters();
+    public Parameters convertToFhir(Interchange interchange) {
+        Patient patient = createPatient("example");
+
+        Reference managingOrganization = createManagingOrganizationReference(interchange);
+        Reference generalPractitioner = createGeneralPractitionerReference(interchange);
+
+        patient.setManagingOrganization(managingOrganization);
+        patient.addGeneralPractitioner(generalPractitioner);
+
+        return createParameters(patient);
     }
 
+    private Parameters createParameters(Resource... resources) {
+        Parameters parameters = new Parameters();
+        for (Resource resource : resources) {
+            parameters.addParameter(new Parameters.ParametersParameterComponent().setResource(resource));
+        }
+        return parameters;
+    }
+
+    private Reference createGeneralPractitionerReference(Interchange interchange) {
+        String gpId = interchange.getGpNameAndAddress().getIdentifier();
+        return new Reference(
+                new Practitioner().setId(gpId)
```

Comment:
Shouldn't this just be a string ref like "Practitioner/<id>" ? The same for organization

Reply:
Output of test patient looked like that:
````
{
  "resourceType": "Parameters",
  "parameter": [
    {
      "resource": {
        "resourceType": "Patient",
        "identifier": [
          {
            "value": "example id"
          }
        ],
        "generalPractitioner": [
          {
            "reference": "Practitioner/2750922,295"
          }
        ],
        "managingOrganization": {
          "reference": "Organization/XX1"
        }
      }
    }
  ]
}
````